src/Compilers/CSharp/Test/Semantic/Semantics/InlineArrayTests.cs

```diff
@@ -0,0 +1,6195 @@
+            CompileAndVerify(comp, symbolValidator: verify, sourceSymbolValidator: verify).VerifyDiagnostics();
+
+            void verify(ModuleSymbol m)
+            {
+                var buffer = m.GlobalNamespace.GetTypeMember("Buffer");
+
+                Assert.True(buffer.HasInlineArrayAttribute(out int length));
+                Assert.Equal(10, length);
+                Assert.False(buffer.TryGetInlineArrayElementType().HasType);
+            }
+        }
+
+        [Fact]
+        public void InlineArrayType_13_Enum()
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(10)]
+enum Buffer
+{
+    _element0
+}
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All, AllowMultiple = false)]
+    public sealed class InlineArrayAttribute : Attribute
+    {
+        public InlineArrayAttribute (int length)
+        {
+            Length = length;
+        }
+
+        public int Length { get; }
+    }
+}
+";
+            var comp = CreateCompilation(src, options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+            CompileAndVerify(comp, symbolValidator: verify, sourceSymbolValidator: verify).VerifyDiagnostics();
+
+            void verify(ModuleSymbol m)
+            {
+                var buffer = m.GlobalNamespace.GetTypeMember("Buffer");
+
+                Assert.True(buffer.HasInlineArrayAttribute(out int length));
+                Assert.Equal(10, length);
+                Assert.False(buffer.TryGetInlineArrayElementType().HasType);
+            }
+        }
+
+        [Fact]
+        public void InlineArrayType_14_Delegate()
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(10)]
+delegate void Buffer();
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All, AllowMultiple = false)]
+    public sealed class InlineArrayAttribute : Attribute
+    {
+        public InlineArrayAttribute (int length)
+        {
+            Length = length;
+        }
+
+        public int Length { get; }
+    }
+}
+";
+            var comp = CreateCompilation(src, options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+            CompileAndVerify(comp, symbolValidator: verify, sourceSymbolValidator: verify).VerifyDiagnostics();
+
+            void verify(ModuleSymbol m)
+            {
+                var buffer = m.GlobalNamespace.GetTypeMember("Buffer");
+
+                Assert.True(buffer.HasInlineArrayAttribute(out int length));
+                Assert.Equal(10, length);
+                Assert.False(buffer.TryGetInlineArrayElementType().HasType);
+            }
+        }
+
+        [Fact]
+        public void InlineArrayType_15_Interface()
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(10)]
+interface Buffer
```

Comment:
Consider testing also `record`s (e.g., with synthesized fields).

Reply:
> Consider testing also records (e.g., with synthesized fields).

Captured in the test plan